### DISCUSSION TITLE
Nextflow submitted AWS Batch user value contains invalid characters

### DISCUSSION THREAD

**Discussion by watsonar** [nextflow-io/nextflow#2900] [Q&A] [answered]
Hello, I'm trying to run a Nextflow workflow with AWS Batch, but all my jobs fail with the following error:
```
The user value contains invalid characters. Enter a value that matches the pattern ^([a-z0-9_][a-z0-9_-]{0,30})$
```

The workflow is a small test example from a Swiss Institute of Bionformatics course, with code on GitHub [here](https://github.com/biocorecrg/SIB_course_nextflow_Nov_2021/tree/main/nextflow/test3). I'm running it with the command `nextflow run test3.nf -with-docker -profile cloud`.

In the container section of the job details page, the user value is `$(id`, which I'm assuming is the cause of this error. I think this is happening because in my nexflow.config file, I have the following line:
```
containerOptions = { workflow.containerEngine == "docker" ? '-u $(id -u):$(id -g)': null}
```

I believe this line is supposed to avoid having Docker run as root, though if I remove the line completely my jobs never get past the "Runnable" state in AWS Batch.

I would greatly appreciate any insight into how to solve this error!

**Comment by bentsherman** [comment]
I think you have to escape the dollar signs: `'-u \$(id -u):\$(id -g)'`

**Comment by watsonar** [comment, marked as answer]
It turns out that the EC2 instance role we were using had limited permissions to S3. Changing those permissions allowed us to run a similar workflow without changing the aforementioned line. Posting in case this helps anyone in the future!
